File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const map = new Map();
    
    for(const [name, kind] of clothes){
        map.set(kind, (map.get(kind) || 0) + 1);
    }
    
    let count = 1;
    for(const value of map.values()){
        count *= (value + 1);
    }

    return count - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    c = {}
    
    for n, t in clothes :
        if t in c :
            c[t] += 1
        else :
            c[t] = 1
    
    
    answer = 1
    for t in c :
        answer *= c[t] + 1
            
    return answer - 1